ironforgedbot/decorators.py

```diff
@@ -0,0 +1,45 @@
+import functools
+import logging
+
+import discord
+
+from ironforgedbot.common.helpers import validate_member_has_role
+
+logger = logging.getLogger(__name__)
+
+
+def require_role(role_name: str):
```

Comment:
Why not keep using decorator package? Can get rid of this 3-way wrap

Reply:
The fewer dependencies the better imo, especially if that is the only benefit the package is giving us. Gives us tighter control with testing rolling it ourselves too.

Reply:
By that logic we should get rid of logger and other dependencies :) I'm on-board with avoiding node-style dependencies hell, but if it simplifies code I'd argue it's better to use it. Not only it's easier to read, but it won't mask stack-traces that much.

Reply:
Logger is a standard library, but I get your point! :grin: 

We should strive to have clean code, but also reduce bloat. I think if we had 4 or 5 decorators we could look into simplifying with a third party package, but as it stands I don't think we are going to need many (if any) more.

The package you were using also hasn't had a release since 2020. While not the end of the world, it doesn't fill me with confidence.